primitives/forest-verifier/src/lib.rs

```diff
@@ -34,6 +34,11 @@ where
         challenges: &[Self::Challenge],
         proof: &Self::Proof,
     ) -> Result<BTreeSet<Self::Challenge>, DispatchError> {
+        // Check that `challenges` is not empty.
+        if challenges.is_empty() {
+            return Err("No challenges provided.".into());
+        }
+
```

Comment:
I see why you made this. The reason for not checking, it, IIRC, was that technically, if there are no challenges, any proof is valid.